### PR TITLE
Search / Filter improvements for the Monitor

### PR DESCRIPTION
**Pull request by MartinDomig**
The Monitor allows to search for
- Groups (as before)
- Name of the vessel
- Name of central body of the vessel
- Partial words (you don't have to type out minmus all the way to
  the end to find your vessels orbiting it)
- Search is case insensitive now

This could also be a first step towards adding vessel type filter buttons (#186). Now that the filter is always visible in the montior, it would be the ideal place to put those icons: make the vessel type icons aligned bottom left, put the search text input next to it.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 * Stock antennas can now be controlled by automation (Yaar Podshipnik)
 * Devices shown in the device manager are now sorted (Sir Mortimer)
 * Fixed the EC issue when accelerating to extremely fast time warp while a ship is in shadow (Sir Mortimer)
-* Iproved vessel searching in monitor: you can search for the name of the central body and the vessel name (Sir Mortimer)
+* Iproved vessel search in monitor: you can search for the name of the central body and the vessel name (Sir Mortimer)
```
Comment (PiezPiedPy) [inline CHANGELOG.md]: Typo 😛 Iproved ? Improved

File: src/UI/Monitor.cs [hunk 2]
```diff
@@ -312,16 +304,55 @@ void Render_menu(Vessel v)
...
+
+		Texture2D getVesselTypeIcon(VesselType type, Boolean disabled = false)
+		{
+			// TODO use proper ship type icons here. There should be a way to fetch those from
+			// KSP, but I don't know how or where to get them
+			switch(type) {
+				// this is what it will look like once someone has found the correct icons...
+				// case VesselType.Base: return disabled ? Icons.base_grey : Icons.base_white;
+
+				case VesselType.Base: return disabled ? Icons.battery_yellow : Icons.battery_white;
```
Comment (MartinDomig) [inline src/UI/Monitor.cs]: I need someone to actually provide the correct icons here, these are just placeholders. However, as I currently a) don't see any icons at all in my own builds (mac...) and b) wouldn't know where to get those icons from (I'm sure they're somewhere readily available within the depths of the KSP/Unity API...) I can't do that myself.

Help, anyone?

Comment (MartinDomig) [reply src/UI/Monitor.cs]: This would fix #186

Comment (PiezPiedPy) [reply src/UI/Monitor.cs]: I'll have a go later tonight if I have some spare time but will more likely be Saturday

File: CHANGELOG.md (hunk 1 above)
Comment (MartinDomig) [reply CHANGELOG.md]: That's the least of my problems ;)

Comment (PiezPiedPy) [reply CHANGELOG.md]: lol

**Pull request merged**